### PR TITLE
Implement CollationSupport and CollatorCache in ars-collections

### LINKED ISSUE #518: task: Implement CollationSupport and CollatorCache in ars-collections
**Wave 4 (advanced)** | **Points: 2** | **Epic: #53 (Collections)**

## Goal

Implement locale-aware collation integration for collection types, connecting `ars-i18n::StringCollator` to `FilteredCollection` and `SortedCollection`.

## Layer

Subsystem

## Framework

None

## Test tier

Unit

## Depends on

- #136 (StringCollator with ICU4X backend — defines the public API)
- #84 (FilteredCollection and SortedCollection)

## Spec refs

- `spec/foundation/06-collections.md` §7.3 "Locale-Aware Collation" (L4556)
- `spec/foundation/06-collections.md` §7.3.1 "StringCollator Integration with Collection Trait" (L4560)

## Files to create/modify

- `crates/ars-collections/src/sorted_collection.rs` (add collation traits, feature-gated)

## Scope

All behind `#[cfg(feature = "i18n")]`:

- `CollationTarget` trait with associated `Item` type
- `CollationSupport` trait with `with_collation()` method
- `impl CollationTarget` for `StaticCollection<T>`, `TreeCollection<T>`, `FilteredCollection<'_, T, C>`
- `impl CollationSupport` for `&StaticCollection<T>`, `&TreeCollection<T>`, `&FilteredCollection<'_, T, C>`
- `CollatorCache` struct keyed by `(Locale, CollationStrength)` with `get_or_create()` method

## Tests to add first

- `CollationSupport` produces a `SortedCollection` with locale-correct ordering
- `CollatorCache::get_or_create` returns cached instance on second call
- `with_collation` works on `StaticCollection`, `TreeCollection`, and `FilteredCollection`

## Acceptance criteria

- All collation types compile only under `i18n` feature
- `CollationSupport` produces `SortedCollection` wrappers
- `CollatorCache` avoids redundant `StringCollator` creation
- All tests TDD-first, zero warnings

## Spec impact

No spec change required.

### PR DESCRIPTION
**Pull request by fogodev**
## Summary
- Implements locale-aware collation integration for `ars-collections`, connecting `ars-i18n::StringCollator` to `StaticCollection`, `TreeCollection`, and `FilteredCollection` via `CollationTarget` and `CollationSupport` traits
- Adds `CollatorCache` keyed by `(Locale, CollationStrength)` to avoid repeated ICU4X collator creation
- All types feature-gated behind `#[cfg(feature = "i18n")]`, organized as a `sorted_collection/collation` submodule
- Updates spec §7.3.1 with blanket `&T` impl, missing `CollationTarget` impls, explicit lifetime params, `const new()`, and `Default`/`Debug` for `CollatorCache`

Closes #518

## Test plan
- [x] 9 collation tests covering `with_collation` on all 3 collection types (flat, tree, filtered), `CollatorCache` caching + miss behavior, `Default` impl, empty collection edge case, and sectioned collection section-preservation
- [x] `cargo test -p ars-collections --features i18n` — 467 passed, 0 failed
- [x] `cargo test -p ars-collections` — all existing tests pass without `i18n` feature
- [x] `cargo clippy -p ars-collections --features i18n -- -D warnings` — zero warnings
- [x] `cargo clippy -p ars-collections -- -D warnings` — zero warnings
- [x] `cargo llvm-cov` — 100% line coverage on all new collation code
- [x] `cargo xci` — exit code 0

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/ars-collections/src/sorted_collection/collation.rs [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+/// [`StringCollator`]). Wraps the collection in a [`SortedCollection`]
+/// using the provided collator for locale-correct string ordering.
+pub trait CollationSupport: Sized + CollationTarget {
+    /// The output type after applying collation (typically a [`SortedCollection`] wrapper).
+    type Output;
+
+    /// Apply locale-aware sorting using the given collator and text extraction function.
+    ///
+    /// `text_fn` extracts the sortable text from each item.
+    fn with_collation<F>(self, collator: StringCollator, text_fn: F) -> Self::Output
```
[the remark this answers is not in the document]
Comment (fogodev) [reply crates/ars-collections/src/sorted_collection/collation.rs]: Fixed in `99a8099`. My earlier PR comment dismissing this was wrong — the by-value signature was a real composability flaw, not intentional API separation.

Since `SortedCollection::new` consumes the comparator eagerly, `&StringCollator` only needs to live through construction — no lifetime pollution into the output. Also dropped the unnecessary `'static` bound on `text_fn` for the same reason. New test `collator_cache_composes_with_collation_support` proves cache → trait composition works end-to-end.

File: crates/ars-collections/src/sorted_collection/collation.rs [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+    type Item;
+}
+
+/// Blanket impl so `&StaticCollection<T>` etc. satisfy the [`CollationTarget`]
+/// supertrait required by [`CollationSupport`] without duplicating impls.
+impl<T: CollationTarget> CollationTarget for &T {
+    type Item = T::Item;
+}
+
+impl<T: CollectionItem + Clone> CollationTarget for StaticCollection<T> {
```
[the remark this answers is not in the document]
Comment (fogodev) [reply crates/ars-collections/src/sorted_collection/collation.rs]: Fixed in `bab880f`. `with_collation` uses the caller-supplied `text_fn` closure for text extraction and never calls `CollectionItem::key()` or `CollectionItem::text_value()` — the bound was accidental spec over-specification.

Relaxed all 6 `CollationTarget` and `CollationSupport` impls from `T: CollectionItem + Clone` to `T: Clone`, matching what `StaticCollection`, `SortedCollection`, and `Collection<T>` themselves require. Spec §7.3.1 updated to match.
